Framework/PythonInterface/mantid/plots/datafunctions.py

```diff
@@ -1039,3 +1040,26 @@ def update_colorbar_scale(figure, image, scale, vmin, vmax):
                                              "as the range between min value and max value is too large")
         figure.subplots_adjust(wspace=0.5, hspace=0.5)
         figure.colorbar(image, ax=figure.axes, ticks=locator, pad=0.06)
+
+def get_images_from_figure(figure):
+    """Return a list of images in the given figure excluding any colorbar images"""
+    axes = figure.get_axes()
+
+    images = []
+    for ax in axes:
+        images += ax.images + [col for col in ax.collections if isinstance(col, QuadMesh)
+                               or isinstance(col, Poly3DCollection)]
+
+    # remove any colorbar images
+    for img in images:
+        if img.colorbar:
+            images.remove(img.colorbar.solids)
```

Comment:
Sorry didn't see this the first time, but from what I can see we are modifying the list while iterating over it (I think), which should be undefined behaviour. It appears to work, but might be best to iterate over a copy, or use list comprehension to make a new list.

Reply:
Ah good spot, I had noticed that, I'll change it